concourse/pipelines/guest-package-build.jsonnet

```diff
@@ -0,0 +1,1177 @@
+local underscore(input) = std.strReplace(input, '-', '_');
+
+local buildpackagejob = {
+  local tl = self,
+
+  package:: error 'must set package in buildpackagejob',
+  gcs_dir:: tl.package,
+  repos:: error 'must set repos in buildpackagejob',
+  uploads:: error 'must set uploads in buildpackagejob',
+  extra_tasks:: [],
+
+  // Start of output.
+  name: 'build-' + tl.package,
+  plan: [
+    {
+      in_parallel: {
+        steps: [
+          {
+            get: tl.package,
+            trigger: true,
+            params: { skip_download: true },
+          },
+          { get: 'guest-test-infra' },
+        ],
+      },
+    },
+    { load_var: 'commit-sha', file: '%s/.git/ref' % tl.package },
+    {
+      task: 'generate-timestamp',
+      config: {
+        platform: 'linux',
+        image_resource: {
+          type: 'registry-image',
+          source: { repository: 'bash' },
+        },
+        outputs: [{ name: 'timestamp' }],
+        run: {
+          path: '/usr/local/bin/bash',
+          args: [
+            '-c',
+            'timestamp=$((${EPOCHREALTIME/./}/1000)); echo $(($timestamp/1000)) | tee timestamp/timestamp; echo $timestamp | tee timestamp/timestamp-ms',
+          ],
+        },
+      },
+    },
+    { load_var: 'start-timestamp-ms', file: 'timestamp/timestamp-ms' },
+    {
+      task: 'generate-package-version',
+      config: {
+        platform: 'linux',
+        image_resource: {
+          type: 'registry-image',
+          source: { repository: 'alpine/git' },
+        },
+
+        inputs: [{ name: tl.package, path: 'repo' }],
+        outputs: [{ name: 'package-version' }],
+
+        run: {
+          path: 'ash',
+          args: [
+            '-exc',
+            // Ugly way to produce multi-line script. TODO: maybe move to scripts?
```

Comment:
IMO moving to a script makes sense. Added bonus is then you'd then be able to have automated tests for scripts that get inserted into overall config?

Reply:
would love to hear your thoughts on how to test scripts. OK to add this as a followup improvement and not in this PR?

Reply:
Yeah that is fine to be a follow up; that is what I intended but should have stated.

Not sure if ash will work with bash here, but I found one framework that may be interesting: https://github.com/bats-core/bats-core